File: ios/Kwift/com/lightningkite/kwift/observables/shared/ConstantObservableProperty.swift
//Package: com.lightningkite.kwift.observables.shared
//Converted using Kwift2

import Foundation



public class ConstantObservableProperty<T>: ObservableProperty<T> {
    
    public var underlyingValue: T
    
    override public var onChange: NeverEvent<T> { get { return _onChange } set(value) { _onChange = value } }
    override public var value: T {
        get {
            return underlyingValue
        }
    }
    
    public init(underlyingValue: T) {
        self.underlyingValue = underlyingValue
        self._onChange = NeverEvent<T>()
        super.init()
    }
    convenience public init(_ underlyingValue: T) {
        self.init(underlyingValue: underlyingValue)
    }
    private var _onChange: NeverEvent<T>
}
 

File: android/src/main/java/com/lightningkite/kwift/observables/shared/ConstantObservableProperty.kt
package com.lightningkite.kwift.observables.shared

class ConstantObservableProperty<T>(val underlyingValue: T) : ObservableProperty<T>() {
    override val onChange: NeverEvent<T> = NeverEvent<T>()
    override val value: T
        get() = underlyingValue
}
